pkg/scheduler/scheduler.go

```diff
@@ -612,10 +612,15 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 			if klog.V(2).Enabled() {
 				klog.InfoS("Successfully bound pod to node", "pod", klog.KObj(pod), "node", scheduleResult.SuggestedHost, "evaluatedNodes", scheduleResult.EvaluatedNodes, "feasibleNodes", scheduleResult.FeasibleNodes)
 			}
-
 			metrics.PodScheduled(prof.Name, metrics.SinceInSeconds(start))
 			metrics.PodSchedulingAttempts.Observe(float64(podInfo.Attempts))
-			metrics.PodSchedulingDuration.Observe(metrics.SinceInSeconds(podInfo.InitialAttemptTimestamp))
+
+			// We breakdown the pod scheduling duration by attempts (capped to a limit).
+			attempts := podInfo.Attempts
```

Comment:
@logicalhan do we really need to cap? can I just use attempts as is?

Reply:
while I don't think the number of attempts will have a wide range of values, leaving it unbounded is certainly not a good idea.